src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/umbblockgridsortable.directive.js

```diff
@@ -0,0 +1,705 @@
+                    element.querySelectorAll(criteria.trim()).forEach(setupPreventEvent);
+                });
+            }
+            function destroyIgnorerElements(element) {
+                config.ignorerSelector.split(',').forEach(function (criteria) {
+                    element.querySelectorAll(criteria.trim()).forEach(destroyPreventEvent);
+                });
+            }
+            function setupPreventEvent(element) {
+                element.draggable = false
+            }
+            function destroyPreventEvent(element) {
+                element.removeAttribute('draggable');
+            }
+
+
+
+
+
+            let currentContainerElement = containerEl;
+            let currentContainerVM = vm;
+
+            let rqaId = null;
+            let currentItem = null;
+            let currentElement = null;
+            let currentDragElement = null;
+            let currentDragRect = null;
+            let dragX = 0;
+            let dragY = 0;
+
+            function handleDragStart(event) {
+
+                if(currentElement) {
+                    handleDragEnd();
+                }
+
+                event.stopPropagation();
+                event.dataTransfer.effectAllowed = "move";// copyMove when we enhance the drag with clipboard data.
+                event.dataTransfer.dropEffect = "none";// visual feedback when dropped.
+                
+                if(!scrollElement) {
+                    scrollElement = getParentScrollElement(containerEl, true);
+                }
+                
+                const element = event.target.closest(config.itemSelector);
+                
+                currentElement = element;
+                currentDragElement = config.draggableSelector ? currentElement.querySelector(config.draggableSelector) : currentElement;
+                currentDragRect = currentDragElement.getBoundingClientRect();
+                currentItem = vm.getItemOfElement(currentElement);
+                if(!currentItem) {
+                    console.error("Could not find item related to this element.")
+                }
+
+                currentElement.style.transform = 'translateZ(0)';// Solves problem with FireFox and ShadowDom in the drag-image.
+                
+                if (config.dataTransferResolver) {
+                    config.dataTransferResolver(event.dataTransfer, currentItem);
+                }
+
+                if (config.onStart) {
+                    config.onStart({item: currentItem, element: currentElement});
+                }
+
+                window.addEventListener('dragover', handleDragMove);
+                window.addEventListener('dragend', handleDragEnd);
+
+                // We must wait one frame before changing the look of the block.
+                rqaId = requestAnimationFrame(() => {// It should be okay to use the same refId, as the move does not or is okay not to happen on first frame/drag-move.
```

Comment:
Do we need an animation frame here to remove the style if we didn't need a frame to set the style further up the function?

Reply:
Yeah, this is a bit tricky. But we want the browser to capture/screenshot the state of the block before the placeholder class is added. (and the transform one is also a fix for the capture, which can go away after that has happened.)

Reply:
So that is fully intentional :-)